### PR TITLE
feat: Add logging to `Cargo.lock`/`Cargo.toml` hashing

### PR DESCRIPTION
**Pull request by NobodyXu**
There are a few problems in cargo-binstall where sometimes change to `Cargo.lock` does not invalidate the cache.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,6 +28,7 @@
     "@actions/exec": "^1.1.1",
     "@actions/glob": "^0.4.0",
     "@actions/io": "^1.1.3",
+    "github-actions.cache-buildjet": "^0.2.0",
```
Comment (Swatinem) [inline package.json:31]: I guess this is the same as `"@actions/buildjet-cache": "npm:github-actions.cache-buildjet@0.1.2",` above, just using a different name. Either way, that change should be done as its own PR, and not intermingled here.

File: src/config.ts [hunk 2]
```diff
@@ -175,8 +175,8 @@ export class CacheConfig {
           hasher.update(JSON.stringify(parsed));
 
           parsedKeyFiles.push(cargo_manifest);
-        } catch (_e) {
-          // Fallback to caching them as regular file
+        } catch (e) { // Fallback to caching them as regular file
+          core.warning(`Error parsing Cargo.toml manifest, fallback to caching entire file: ${e}`);
```
Comment (Swatinem) [inline src/config.ts:179]: Instead of using a `warning` which creates an annotation (similar to the issue raised in https://github.com/Swatinem/rust-cache/issues/144), does it make sense to use `debug` rather?

**Review by Swatinem — COMMENTED** [review]
I’m a bit unsure if this helps. You say that changes to `Cargo.lock` do not lead to a new cache key, but these fallback paths will consider the entire `Cargo.lock`.

File: package.json (hunk 1 above)
Comment (NobodyXu) [reply package.json]: Thanks, I've removed this.

File: src/config.ts (hunk 2 above)
Comment (NobodyXu) [reply src/config.ts:179]: I think it's better to issue a warning, so that the warning can be seen even if it's not a debug run.

I don't think it will be as bad as #144 since it's a warning not an error.

File: package-lock.json [hunk 3]
```diff
@@ -15,6 +15,7 @@
         "@actions/exec": "^1.1.1",
         "@actions/glob": "^0.4.0",
         "@actions/io": "^1.1.3",
+        "github-actions.cache-buildjet": "^0.2.0",
```
Comment (Swatinem) [inline package-lock.json:18]: pls revert this from the package-lock as well :-)

**Review by Swatinem — APPROVED** [review]

File: package-lock.json (hunk 3 above)
Comment (NobodyXu) [reply package-lock.json]: Thanks, I've reverted this.

**Review by Swatinem — APPROVED** [review]
